### PR TITLE
Add 2d thermo data q-coe

### PR DESCRIPTION
**Pull request by tduretz**
Databases of density of SiO2 (=f(P,T)) and dG of reaction of q-coe is implemented. It also feature an basic support of reaction kinetics based on the Mosenfelder paper (1997).

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -43,7 +43,15 @@ set(SuiteSparse_DIR ${PROJECT_SOURCE_DIR}/lib/suitesparse/install/lib/cmake/suit
 set(HDF5_DIR ${PROJECT_SOURCE_DIR}/lib/hdf5/install/share/cmake)
 FIND_PACKAGE(SuiteSparse CONFIG)
 FIND_PACKAGE(HDF5 CONFIG)
-configure_file("SOURCE/${MODEL}.txt" "${CMAKE_BINARY_DIR}/setup.txt" COPYONLY)
+
+if (EXISTS SOURCE/${MODEL}.txt)
+    configure_file("SOURCE/${MODEL}.txt" "${CMAKE_BINARY_DIR}/setup.txt" COPYONLY)
```
Comment (kulakovri) [inline CMakeLists.txt]: Please merge main branch into this one

File: SOURCE/FlowLaws.c [hunk 2]
```diff
@@ -894,3 +894,30 @@ void ReadDataGSE( mat_prop* mat, params* model, int k, int number, scale* scalin
 /*--------------------------------------------------------------------------------------------------------------------*/
 /*------------------------------------------------------ M-Doodz -----------------------------------------------------*/
 /*--------------------------------------------------------------------------------------------------------------------*/
+
+void ReadDataKinetics( mat_prop* mat, params* model, int k, int number, scale* scaling ) {
+    
+    int success=0;
```
Comment (kulakovri) [inline SOURCE/FlowLaws.c]: I don't understand why do we need this

File: SOURCE/FlowLaws.c [hunk 3]
```diff
@@ -894,3 +894,30 @@ void ReadDataGSE( mat_prop* mat, params* model, int k, int number, scale* scalin
 /*--------------------------------------------------------------------------------------------------------------------*/
 /*------------------------------------------------------ M-Doodz -----------------------------------------------------*/
 /*--------------------------------------------------------------------------------------------------------------------*/
+
+void ReadDataKinetics( mat_prop* mat, params* model, int k, int number, scale* scaling ) {
```
Comment (kulakovri) [inline SOURCE/FlowLaws.c:898]: Number refers to phase_number? Phase_Id?

File: CMakeLists.txt (hunk 1 above)
Comment (kulakovri) [reply CMakeLists.txt]: and resolve conflicts

**Review by kulakovri — CHANGES_REQUESTED** [review]
Also we've increasing repository size with those .dat files by 30-40 Mb
We find a workaround to avoid that

File: SOURCE/FlowLaws.c (hunk 2 above)
Comment (tduretz) [reply SOURCE/FlowLaws.c]: We need this in case we enter several kinetic law (then each law has its own ID)

File: SOURCE/FlowLaws.c (hunk 3 above)
Comment (tduretz) [reply SOURCE/FlowLaws.c:898]: Number correspond to the ID of the kinetic law

**Review by kulakovri — APPROVED** [review]

**Pull request merged**